hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFailureHandling.java

```diff
@@ -200,9 +203,14 @@ public void testSingleObjectDeleteNoPermissionsTranslated() throws Throwable {
     Path path = requireDefaultExternalData(getConfiguration());
     S3AFileSystem fs = (S3AFileSystem) path.getFileSystem(
         getConfiguration());
-    AccessDeniedException aex = intercept(AccessDeniedException.class,
+    Class exceptionClass = AccessDeniedException.class;
+    if (CSEUtils.isCSEEnabled(getEncryptionAlgorithm(
+        getTestBucketName(getConfiguration()), getConfiguration()).getMethod())) {
+      exceptionClass = AWSClientIOException.class;
```

Comment:
why a different exception? It should be access denies in both cases, shouldn't it?

After all, that is what test is meant to do: delete no permissions are translated.

Reply:
so the SDK is raising some exception? does it have the same semantics as Access Denied? that is: should it be remapped?

Reply:
Yes, The semantics is same "Caused by: software.amazon.encryption.s3.S3EncryptionClientException: Access Denied (Service: S3, Status Code: 403,"

Reply:
can you show me the full stack? 403 normally maps to AccessDeniedException, and it'd be good to keep the same.  AWSClientIOException is just our "something failed" if there's nothing else

1. Add `maybeTranslateEncryptionClientException()` in `ErrorTranslation` to look at the exception, if the classname string value matches "software.amazon.encryption.s3.S3EncryptionClientException" then map to AccessDeniedException
2. call that to S3AUtils.translateException just before the ` // no custom handling.` bit

It may seem bad, but look at `maybeExtractChannelException()` and other bits to see worse.

Reply:
The following is the complete stacktrace

```
[ERROR] testSingleObjectDeleteNoPermissionsTranslated(org.apache.hadoop.fs.s3a.ITestS3AFailureHandling)  Time elapsed: 26.964 s  <<< ERROR!
org.apache.hadoop.fs.s3a.AWSClientIOException: delete on <>: software.amazon.encryption.s3.S3EncryptionClientException: Access Denied (Service: S3, Status Code: 403,: Access Denied (Service: S3, Status Code: 403, )
	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:228)
	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:124)
	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:163)
	at org.apache.hadoop.fs.s3a.S3AFileSystem$OperationCallbacksImpl.deleteObjectAtPath(S3AFileSystem.java:2637)
	at org.apache.hadoop.fs.s3a.impl.DeleteOperation.deleteObjectAtPath(DeleteOperation.java:383)
	at org.apache.hadoop.fs.s3a.impl.DeleteOperation.execute(DeleteOperation.java:234)
	at org.apache.hadoop.fs.s3a.impl.DeleteOperation.execute(DeleteOperation.java:70)
	at org.apache.hadoop.fs.s3a.impl.ExecutingStoreOperation.apply(ExecutingStoreOperation.java:76)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.invokeTrackingDuration(IOStatisticsBinding.java:547)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:528)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.trackDuration(IOStatisticsBinding.java:449)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.deleteWithoutCloseCheck(S3AFileSystem.java:3595)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.delete(S3AFileSystem.java:3571)
	at org.apache.hadoop.fs.s3a.ITestS3AFailureHandling.lambda$testSingleObjectDeleteNoPermissionsTranslated$2(ITestS3AFailureHandling.java:207)
	at org.apache.hadoop.test.LambdaTestUtils.intercept(LambdaTestUtils.java:500)
	at org.apache.hadoop.test.LambdaTestUtils.intercept(LambdaTestUtils.java:386)
	at org.apache.hadoop.fs.s3a.ITestS3AFailureHandling.testSingleObjectDeleteNoPermissionsTranslated(ITestS3AFailureHandling.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: software.amazon.encryption.s3.S3EncryptionClientException: Access Denied (Service: S3, Status Code: 403,
	at software.amazon.encryption.s3.S3EncryptionClient.deleteObject(S3EncryptionClient.java:377)
	at org.apache.hadoop.fs.s3a.impl.S3AStoreImpl.lambda$deleteObject$5(S3AStoreImpl.java:677)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.invokeTrackingDuration(IOStatisticsBinding.java:547)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:528)
	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:468)
	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:431)
	at org.apache.hadoop.fs.s3a.impl.S3AStoreImpl.deleteObject(S3AStoreImpl.java:668)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.deleteObject(S3AFileSystem.java:3223)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.deleteObjectAtPath(S3AFileSystem.java:3248)
	at org.apache.hadoop.fs.s3a.S3AFileSystem$OperationCallbacksImpl.lambda$deleteObjectAtPath$0(S3AFileSystem.java:2638)
	at org.apache.hadoop.fs.s3a.Invoker.lambda$once$0(Invoker.java:165)
	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:122)
	... 30 more
Caused by: software.amazon.awssdk.services.s3.model.S3Exception: Access Denied (Service: S3, Status Code: 403, 
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleErrorResponse(AwsXmlPredicatedResponseHandler.java:156)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleResponse(AwsXmlPredicatedResponseHandler.java:108)
```

Reply:
Yes, error translation should do the job.